deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py

```diff
@@ -94,12 +103,214 @@ def get_data_sample(self) -> dict:
                     'sharedEventID': str(uuid4())
                 }
             ]
-        }
+        })
+
+
+class VPCDataGenerator(DataGenerator):
+    BASE_PATH = f'{cons.AWS_LOGS}/{cons.RANDOM_ACCOUNT_ID}/{cons.VPC_FLOW_LOGS}/{cons.US_EAST_1_REGION}/'
+    BASE_FILE_NAME = f'{cons.RANDOM_ACCOUNT_ID}_{cons.VPC_FLOW_LOGS}_{cons.US_EAST_1_REGION}_'
+
+    def get_filename(self, prefix=None, **kwargs) -> str:
+        """Return the filename in the cloudtrail format
+        <prefix>/AWSLogs/<suffix>/<organization_id>/<account_id>/vpcflowlogs/<region>/<year>/<month>/<day>
+        """
+        now = datetime.now()
+        path = f'{self.BASE_PATH}{now.strftime(cons.PATH_DATE_FORMAT)}/'
+        name = (
+            f'{self.BASE_FILE_NAME}{cons.FLOW_LOG_ID}_{now.strftime(cons.FILENAME_DATE_FORMAT)}_{abs(hash(now))}'
+            f'{cons.LOG_EXT}'
+        )
+
+        return f'{path}{name}'
+
+    def get_data_sample(self) -> str:
+        data = [
+            [
+                "version", "account-id", "interface-id", "srcaddr", "dstaddr", "srcport", "dstport", "protocol",
+                "packets", "bytes", "start", "end", "action", "log-status"
+            ]
+        ]
+
+        for _ in range(5):
+            data.append(
+                [
+                    "2", cons.RANDOM_ACCOUNT_ID, get_random_interface_id(), get_random_ip(), get_random_ip(),
+                    get_random_port(), get_random_port(), "6", "39", "4698", "1622505433", "1622505730", "ACCEPT", "OK"
+                ]
+            )
+        buffer = StringIO()
+        csv.writer(buffer, delimiter=" ").writerows(data)
+
+        return buffer.getvalue()
```

Comment:
I had to cherry-pick the commit 6d77977ea75b102fe0bd34e97f0ca7fcd1738c5a, because this adds the possibility of generating data in CSV format. Needed to for ALB, CLB, and NLB too.

Reply:
Great! :raised_hands: